raster/src/main/scala/geotrellis/raster/rasterize/polygon/FractionalRasterizer.scala

```diff
@@ -0,0 +1,185 @@
+package geotrellis.raster.rasterize.polygon
+
+import geotrellis.raster._
+import geotrellis.raster.rasterize._
+import geotrellis.vector._
+
+import com.vividsolutions.jts.geom.Envelope
+import spire.syntax.cfor._
+
+import scala.collection.mutable
+import scala.math.{min, max, ceil, floor, abs}
+
+
+object FractionalRasterizer {
+
+  private type Segment = (Double, Double, Double, Double)
+
+  private def polygonToEdges(poly: Polygon, re: RasterExtent): Seq[Segment] = {
+
+    val arrayBuffer = mutable.ArrayBuffer.empty[Segment]
+
+    /** Find the outer ring's segments */
+    val coords = poly.jtsGeom.getExteriorRing.getCoordinates
+    cfor(1)(_ < coords.length, _ + 1) { ci =>
+      val coord1 = coords(ci - 1)
+      val coord2 = coords(ci)
+
+      val col1 = re.mapXToGridDouble(coord1.x)
+      val row1 = re.mapYToGridDouble(coord1.y)
+      val col2 = re.mapXToGridDouble(coord2.x)
+      val row2 = re.mapYToGridDouble(coord2.y)
+
+      val segment =
+        if (col1 < col2) (col1, row1, col2, row2)
+        else (col2, row2, col1, row1)
+
+      arrayBuffer += segment
+    }
+
+    /** Find the segments for the holes */
+    cfor(0)(_ < poly.numberOfHoles, _ + 1) { i =>
+      val coords = poly.jtsGeom.getInteriorRingN(i).getCoordinates
+      cfor(1)(_ < coords.length, _ + 1) { ci =>
+        val coord1 = coords(ci - 1)
+        val coord2 = coords(ci)
+
+        val col1 = re.mapXToGridDouble(coord1.x)
+        val row1 = re.mapYToGridDouble(coord1.y)
+        val col2 = re.mapXToGridDouble(coord2.x)
+        val row2 = re.mapYToGridDouble(coord2.y)
+
+        val segment =
+          if (col1 < col2) (col1, row1, col2, row2)
+          else (col2, row2, col1, row1)
+
+        arrayBuffer += segment
+      }
+    }
+
+    arrayBuffer
+  }
+
+  private def renderEdge(
+    edge: Segment,
+    re: RasterExtent,
+    poly: Polygon,
+    set: mutable.Set[(Int, Int)],
+    fn: FractionCallback
+  ): Unit = {
+    // Screen coordinates
+    val (x0, y0, x1, y1) = edge
+    val xmin = min(x0, x1)
```

Comment:
why call these screen coordinates? The convention in the library is that these are "grid coordinates", and we represent them with `col`, `row`, `rowMin`, etc. This differentiates them from `x`, `y`, `xmin` etc which represent map coordinates.